src/Action/Bulk/ToggleAction.php

```diff
@@ -0,0 +1,68 @@
+<?php
+namespace Crud\Action\Bulk;
+
+use Cake\Controller\Controller;
+use Cake\Database\Expression\QueryExpression;
+use Cake\ORM\Query;
+use Crud\Error\Exception\ActionNotConfiguredException;
+
+/**
+ * Handles Bulk 'Toggle' Crud actions
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ */
+class ToggleAction extends BaseAction
+{
+    /**
+     * Constructor
+     *
+     * @param \Cake\Controller\Controller $Controller Controller instance
+     * @param array $config Default settings
+     * @return void
+     */
+    public function __construct(Controller $Controller, $config = [])
+    {
+        $this->_defaultConfig['actionName'] = 'Toggle';
+        $this->_defaultConfig['messages'] = [
+            'success' => [
+                'text' => 'Value toggled successfully'
+            ],
+            'error' => [
+                'text' => 'Could not toggle value'
+            ]
+        ];
+        return parent::__construct($Controller, $config);
+    }
+
+    /**
+     * Handle a bulk event
+     *
+     * @return \Cake\Network\Response
+     */
+    protected function _handle()
+    {
+        $field = $this->config('field');
+        if (empty($field)) {
+            throw new ActionNotConfiguredException('No field value specified');
+        }
+
+        return parent::_handle();
+    }
+
+    /**
+     * Handle a bulk toggle
+     *
+     * @param \Cake\ORM\Query $query The query to act upon
+     * @return bool
+     */
+    protected function _bulk(Query $query = null)
```

Comment:
I think bulk-toggle is not a good an idea.

In principle it's fine - but in practice it means when the action runs you won't have any idea what it'll actually do since it depends on the db state at the time. For example, accidentally double-submit and it "doesn't work at all". Better to use explicit/idempotent actions.

That said, it's providing a tool (rope) for devs to use. Any other thoughts along the above lines?